### PR TITLE
Drivetrain

### PR DESCRIPTION
**Pull request by leomwilson**

### CODE REVIEW

File: src/main/java/frc/robot/RobotContainer.java [hunk 1]
```diff
@@ -42,6 +43,9 @@ public RobotContainer() {
    * {@link edu.wpi.first.wpilibj2.command.button.JoystickButton}.
    */
   private void configureButtonBindings() {
+    drivetrain.setDefaultCommand(new TeleopTankDrive(drivetrain, 
+    () -> -xbox.getY(Hand.kLeft),
+    () -> xbox.getX(Hand.kRight)));
```
Comment (leomwilson) [inline src/main/java/frc/robot/RobotContainer.java]: [Continuation lines should be indented by four spaces.](https://google.github.io/styleguide/javaguide.html#s4.5.2-line-wrapping-indent)

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.DemandType;
+import com.ctre.phoenix.motorcontrol.NeutralMode;
+import com.ctre.phoenix.motorcontrol.can.TalonFX;
+import com.team6479.lib.subsystems.TankDrive;
+
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants.DrivetrainConstants;
+
+public class Drivetrain extends SubsystemBase implements TankDrive{
```
Comment (leomwilson) [inline src/main/java/frc/robot/subsystems/Drivetrain.java]:
```suggestion
public class Drivetrain extends SubsystemBase implements TankDrive {
```

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    rightFront = new TalonFX(DrivetrainConstants.RIGHT_FRONT);
+    rightBack = new TalonFX(DrivetrainConstants.RIGHT_BACK);
+
+    // set motors to factory defaults on startup
+    leftFront.configFactoryDefault();
+    leftBack.configFactoryDefault();
+    rightFront.configFactoryDefault();
+    rightBack.configFactoryDefault();
+
+    // left Back follows left Front and right Back follows right Front
```
Comment (leomwilson) [inline src/main/java/frc/robot/subsystems/Drivetrain.java]:
```suggestion
    // back motors follow front motors
```
Phrasing is a bit awkward.

**Review by leomwilson — COMMENTED** [review]
Overall quite good, just fix the bits that I mentioned and it'll be merged.

File: src/main/java/frc/robot/Constants.java [hunk 4]
```diff
@@ -16,4 +16,10 @@
  * constants are needed, to reduce verbosity.
  */
 public final class Constants {
+    public static final class DrivetrainConstants {
+        public final static int LEFT_FRONT = 0;
+        public final static int LEFT_BACK = 0;
+        public final static int RIGHT_FRONT = 0;
+        public final static int RIGHT_BACK = 0;
```
Comment (tjquillan) [inline src/main/java/frc/robot/Constants.java]: Might want to prefix these with `MOTOR_` just for descriptiveness

**Review by tjquillan — COMMENTED** [review]
Looks good other than my small nitpicks.

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 5]
```diff
@@ -1,5 +1,5 @@
 /*----------------------------------------------------------------------------*/
-/* Copyright (c) 2019 FIRST. All Rights Reserved.                             */
+/* CopymotorRight (c) 2019 FIRST. All Rights Reserved.                             */
```
Comment (leomwilson) [inline src/main/java/frc/robot/subsystems/Drivetrain.java]:
```suggestion
/* Copyright (c) 2019 FIRST. All Rights Reserved.                             */
```

File: src/main/java/frc/robot/Constants.java [hunk 6]
```diff
@@ -17,9 +17,11 @@
...
 public final class Constants {
     public static final class DrivetrainConstants {
-        public final static int LEFT_FRONT = 0;
-        public final static int LEFT_BACK = 0;
-        public final static int RIGHT_FRONT = 0;
-        public final static int RIGHT_BACK = 0;
+        public final static int MOTOR_LEFT_FRONT = 0;
+        public final static int MOTOR_LEFT_BACK = 0;
+        public final static int MOTOR_RIGHT_FRONT = 0;
+        public final static int MOTOR_RIGHT_BACK = 0;
```
Comment (leomwilson) [inline src/main/java/frc/robot/Constants.java:23]: Class doesn't terminate

**Pull request merged**